lib/switchtec.c

```diff
@@ -736,6 +738,24 @@ static int realloc_log_defs(struct log_defs *defs, int num_modules)
 	return 0;
 }
 
+/**
+ * @brief Parse an integer from a string
+ * @param[in] str  - string to parse
+ * @param[out] val - integer
+ * @return true on success, false on failure
+ */
+static bool parse_int(char *str, int *val)
+{
+	char *endptr;
+
+	*val = strtol(str, &endptr, 0);
```

Comment:
In other recent work we were setting errno=0 and then checking that errno is set after calling strtol.... We should probably do the same here.n

Reply:
Sure, I'll add that.